File: ruby/class_inheritance/manager.rb
require './employee.rb'

class Manager < Employee

  attr_accessor :reports # accessor
  def initialize(start)
    super(start)
    @reports = []
  end

  def job_description
    "Suit"
  end

end

#m = Manager.new(Date.today)
#puts m.start_date
#m.reports <<= "blah"
#puts m.reports

File: class-inheritence/java/src/Manager.java
package class_inheritance;

import java.util.Collection;
import java.util.Date;

/**
 * Manager: TODO
 *
 * @author <a href="mailto:jtroxel@yahoo.com">John Troxel</a>
 */
public class Manager extends Employee {
  Collection<Employee> reports;

  public Manager(Date startDate) {
    super(startDate);
  }

  @Override
  public String jobDescription() {
    return "Suit";
  }

  public Collection<Employee> getReports() {
    return reports;
  }
}
